src/analyzer/analyzerthread.cpp

```diff
@@ -0,0 +1,303 @@
+#include "analyzer/analyzerthread.h"
+
+#ifdef __VAMP__
+#include "analyzer/analyzerbeats.h"
+#include "analyzer/analyzerkey.h"
+#endif
+#include "analyzer/analyzergain.h"
+#include "analyzer/analyzerebur128.h"
+#include "analyzer/analyzerwaveform.h"
+
+#include "library/dao/analysisdao.h"
+
+#include "engine/engine.h"
+
+#include "sources/soundsourceproxy.h"
+#include "sources/audiosourcestereoproxy.h"
+
+#include "util/db/dbconnectionpooler.h"
+#include "util/db/dbconnectionpooled.h"
+#include "util/logger.h"
+#include "util/timer.h"
+
+
+namespace {
+
+mixxx::Logger kLogger("AnalyzerThread");
+
+// Analysis is done in blocks.
+// We need to use a smaller block size, because on Linux the AnalyzerThread
+// can starve the CPU of its resources, resulting in xruns. A block size
+// of 4096 frames per block seems to do fine.
+constexpr mixxx::AudioSignal::ChannelCount kAnalysisChannels = mixxx::kEngineChannelCount;
+constexpr SINT kAnalysisFramesPerBlock = 4096;
+const SINT kAnalysisSamplesPerBlock =
+        kAnalysisFramesPerBlock * kAnalysisChannels;
+
+// Maximum frequency of progress updates while busy
+constexpr std::chrono::milliseconds kBusyProgressInhibitDuration(100);
+
+} // anonymous namespace
+
+AnalyzerThread::AnalyzerThread(
+        int id,
+        mixxx::DbConnectionPoolPtr pDbConnectionPool,
+        UserSettingsPointer pConfig,
+        AnalyzerMode mode)
+        : WorkerThread(QString("AnalyzerThread %1").arg(id)),
+          m_id(id),
+          m_pDbConnectionPool(std::move(pDbConnectionPool)),
+          m_pConfig(std::move(pConfig)),
+          m_mode(mode),
+          m_sampleBuffer(kAnalysisSamplesPerBlock),
+          m_emittedState(AnalyzerThreadState::Void),
+          // The first signal should always be emitted
+          m_lastBusyProgressEmittedAt(Clock::now() - kBusyProgressInhibitDuration) {
+    m_nextTrack.setValue(TrackPointer());
+    // The types are registered multiple times although once would be sufficient
+    qRegisterMetaType<AnalyzerThreadState>();
```

Comment:
Because we don't need to if we want to register it with the type's canonical name: http://doc.qt.io/archives/qt-4.8/qmetatype.html#qRegisterMetaType-2